### PR TITLE
Change permission error to Direct Message w updated information

### PR DESCRIPTION
**Pull request by TheStonedTurtle**
Also fixes an issue where the bot responds to invalid commands

#### Direct Message

Upon successfully sending this DM the triggering message will be deleted from the chat

~~Since all custom commands can be ran via DMing the bot the results of the the command have been added to the message. This is to prevent the user from having to resend the same command again as a DM.~~

#### Channel Message (fallback)

This is only sent if the DM fails.

### CODE REVIEW

File: lib/commands.js [hunk 1]
```diff
@@ -19,16 +19,27 @@ function hasAdminPermissions (member) {
...
+    errorMessage += fallbackMessage
+    errorMessage += message.channel ? ' inside ' + message.channel : ''
+
+    fallbackMessage += ' here'
+  } else {
+    fallbackMessage = 'You need to be in a guild channel for this command to work'
+    errorMessage += fallbackMessage
+  }
+
+  errorMessage += '\n\nCommand: **' + message.content + '**'
```
Comment (deathbeam) [inline lib/commands.js]: Use .addField instead of this

File: lib/commands.js [hunk 2]
```diff
@@ -19,16 +19,27 @@ function hasAdminPermissions (member) {
...
 
-  return message.channel.send({
+  return sendDM(message.author, {
     embed: createEmbed()
       .setTitle('Failed to execute command')
       .setDescription(errorMessage)
       .setColor(0xFF0000)
-  })
+  }).then(() => message.delete(), () => message.channel.send(fallbackMessage))
```
Comment (deathbeam) [inline lib/commands.js]: Do not delete the message, and for fallback use Promise.catch

File: lib/commands.js [hunk 3]
```diff
@@ -363,14 +374,14 @@ module.exports = (message, command, args) => {
       break
     }
     default: {
-      if (!canExecuteCustomCommand(message)) {
-        return noPermissions(message)
-      }
-
       const value = db.get(command)
 
       if (value) {
```
Comment (deathbeam) [inline lib/commands.js]: flip this:

if (!value) {
return 
}

if (!canExecuteCustomCommand(message))
{
return
}

etc etc

File: lib/commands.js [hunk 4]
```diff
@@ -19,16 +19,26 @@ function hasAdminPermissions (member) {
...
 
-  return message.channel.send({
+  return sendDM(message.author, {
     embed: createEmbed()
       .setTitle('Failed to execute command')
       .setDescription(errorMessage)
+      .addField('Command', message.content)
       .setColor(0xFF0000)
-  })
+  }).catch(err => message.channel.send(fallbackMessage))
```
Comment (deathbeam) [inline lib/commands.js]: you are not doing anything with err, so either log it like we do everywhere else or just use ()

File: lib/commands.js [hunk 5]
```diff
@@ -363,15 +373,16 @@ module.exports = (message, command, args) => {
       break
     }
     default: {
+      const value = db.get(command)
+      if (!value) {
```
Comment (deathbeam) [inline lib/commands.js]: add empty line above

File: lib/commands.js [hunk 6]
```diff
@@ -20,15 +20,18 @@ function hasAdminPermissions (member) {
 
 function noPermissions (message) {
   const errorMessage = message.member
-    ? 'You do not have enough permissions to run this command'
-    : 'You need to be in guild channel for this command to work'
+    ? 'You can\'t run this command '
+    : 'You need to be in a guild channel for this command to work'
 
-  return message.channel.send({
+  const place = message.channel && message.member ? 'inside ' + message.channel : ''
```
Comment (deathbeam) [inline lib/commands.js]: use here instead of ''.

Comment (deathbeam) [inline lib/commands.js]: add space before inside

File: lib/commands.js [hunk 7]
```diff
@@ -20,15 +20,18 @@ function hasAdminPermissions (member) {
 
 function noPermissions (message) {
   const errorMessage = message.member
-    ? 'You do not have enough permissions to run this command'
-    : 'You need to be in guild channel for this command to work'
+    ? 'You can\'t run this command '
```
Comment (deathbeam) [inline lib/commands.js]: remove space at the end

File: lib/commands.js (hunk 6 above)
Comment (TheStonedTurtle) [reply lib/commands.js]: This will add `here` to the guild message, which seems wrong?

Comment (deathbeam) [reply lib/commands.js]: No, that is fine

Comment (deathbeam) [reply lib/commands.js]: Just do it :d

File: lib/commands.js [hunk 8]
```diff
@@ -20,15 +20,18 @@ function hasAdminPermissions (member) {
...
+
+  return sendDM(message.author, {
     embed: createEmbed()
       .setTitle('Failed to execute command')
-      .setDescription(errorMessage)
+      .setDescription(errorMessage + place)
+      .addField('Command', message.content)
       .setColor(0xFF0000)
-  })
+  }).catch(() => message.channel.send(errorMessage + ' here'))
```
Comment (deathbeam) [inline lib/commands.js]: use errorMessage + place here too (just concat it before)

**Pull request merged**